doc/source/user_guide/scale.rst

```diff
@@ -0,0 +1,404 @@
+.. _scale:
+
+*************************
+Scaling to large datasets
+*************************
+
+Pandas provides data structures for in-memory analytics, which makes using pandas
+to analyze larger than memory datasets somewhat tricky.
+
+This document provides a few recommendations for scaling your analysis to larger datasets.
+It's a complement to :ref:`enhancingperf`, which focuses on speeding up analysis
+for datasets that fit in memory.
+
+But first, it's worth considering *not using pandas*. Pandas isn't the right
+tool for all situations. If you're working with very large datasets and a tool
+like PostgreSQL fits your needs, then you should probably be using that.
+Assuming you want or need the expressivity and power of pandas, let's carry on.
+
+.. ipython:: python
+
+   import pandas as pd
+   import numpy as np
+
+.. ipython:: python
+   :suppress:
+
+   from pandas.util.testing import _make_timeseries
+
+
+Use more efficient file formats
+-------------------------------
+
+Depending on your workload, data loading may be a bottleneck. In these cases you
+might consider switching from a slow format like CSV to a faster format like
+Parquet. Loading from a file format like Parquet will also require less memory
+usage, letting you load larger datasets into pandas before running out of
+memory.
+
+.. ipython:: python
+   :suppress:
+
+   # Make a random in-memory dataset
+   ts = _make_timeseries(freq="30S", seed=0)
+   ts.to_csv("timeseries.csv")
+   ts.to_parquet("timeseries.parquet")
+
+For example, suppose we have a dataset like the following::
+
+                          id      name         x         y
+   timestamp
+   2000-01-01 00:00:00  1029   Michael  0.278837  0.247932
+   2000-01-01 00:00:30  1010  Patricia  0.077144  0.490260
+   2000-01-01 00:01:00  1001    Victor  0.214525  0.258635
+   2000-01-01 00:01:30  1018     Alice -0.646866  0.822104
+   2000-01-01 00:02:00   991       Dan  0.902389  0.466665
+   ...                   ...       ...       ...       ...
+   2000-12-30 23:58:00   992     Sarah  0.721155  0.944118
+   2000-12-30 23:58:30  1007    Ursula  0.409277  0.133227
+   2000-12-30 23:59:00  1009    Hannah -0.452802  0.184318
+   2000-12-30 23:59:30   978     Kevin -0.904728 -0.179146
+   2000-12-31 00:00:00   973    Ingrid -0.370763 -0.794667
+
+That dataset has been stored on disk as CSV and Parquet. We want to
+compare the performance of reading those two formats.
+
+.. ipython:: python
+
+   col = "timestamp"
+   %time _ = pd.read_csv("timeseries.csv", index_col=col, parse_dates=[col])
+
+.. ipython:: python
+
+   %time _ = pd.read_parquet("timeseries.parquet")
+
+Notice that parquet gives higher performance for reading (and writing), both
+in terms of speed and lower peak memory usage. See :ref:`io` for more.
```

Comment:
Maybe link to the section in io.rst that compares the performance of different formats?